core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -97,6 +97,10 @@ private[deploy] class Worker(
   private val APP_DATA_RETENTION_SECONDS =
     conf.getLong("spark.worker.cleanup.appDataTtl", 7 * 24 * 3600)
 
+  // Whether or not cleanup the non-shuffle files on executor finishes.
+  private val CLEANUP_NON_SHUFFLE_FILES_ENABLED =
+    conf.getBoolean("spark.worker.cleanup.nonShuffleFiles.enabled", true)
```

Comment:
Should we document this configuration at https://github.com/apache/spark/blob/master/docs/spark-standalone.md ?

I can't imagine a great reason for users wanting to disable this and believe that this is functioning primarily as a feature-flag to give us an escape-hatch in case of regressions, so maybe it's not a huge deal to leave it undocumented?

Reply:
Is it possible that a user want to cleanup non-shuffle files but don't want to cleanup the whole application directories(may still make use of the shuffle data)? For this specific use case, we should not make `spark.worker.cleanup.enabled = true` a precondition of `spark.worker.cleanup.nonShuffleFiles.enabled = true`. Otherwise, I would be glad to make the change.

Reply:
I think these configurations might actually be independent:

- `spark.worker.cleanup.enabled` controls whether directories from finished applications (i.e. terminated drivers) are cleaned up is a decision which can impact users' ability to debug failed applications. In environments where clusters / underlying physical workers are ephemeral this configuration tends to matter a little less, but in environments with long-lived machines which host many Spark drivers over a long period of time (e.g. on prem) this configuration can matter. For example, I think that setting `spark.worker.cleanup.enabled = true` could mean that completely applications' executor  logs get purged from the underlying host machine after a time period.
- The new configuration being proposed here generally matters only _within a long-running Spark application_ and is kind of orthogonal to the choice to keep or discard failed applications' executor directories. I think that we're only removing files within block managers' nested folders, right? So there should be no user-controlled / user-generated files here, at least in theory. Therefore the impact of this configuration on ordinary users' ability to debug is likely to be small (I think you'd only ever really want access to the files that we're deleting if you're going to hex-dump files to debug corruption issues which trigger segfaults, an unlikely scenario).

Given the kind of different motivations and lifecycles for these cleanup mechanisms and the motivations for different choices of defaults, I'm thinking that it might make sense to rename the new configuration so that we don't confuse users via similar names.

What do you think about something like `spark.storage.removeNonShuffleBlockManagerFilesAfterExecutorDeath`? Really verbose, but a precise description of what's going on. I'm open to better name suggestions (since I tend to pick overly-complicated ones); this is just a conversation-starting strawman.